lib/pattern.dart

```diff
@@ -50,7 +50,7 @@ class _MultiPattern extends Pattern {
       {Iterable<Pattern> this.exclude});
 
   Iterable<Match> allMatches(String str, [int start = 0]) {
-    var _allMatches = [];
+    final _allMatches = <Iterable<Match>>[];
```

Comment:
Or, make it `<Match>[]`, then below change from `_allMatches.add` to `_allMatches.addAll` and remove `expand`.